cuebot/src/main/java/com/imageworks/spcue/dispatcher/DispatchSupportService.java

```diff
@@ -376,12 +376,9 @@ public RunFrame prepareRqdRunFrame(VirtualProc proc, DispatchFrame frame) {
         FrameSet fs = new FrameSet(frame.range);
         int startFrameIndex = fs.index(frameNumber);
         String frameSpec = fs.getChunk(startFrameIndex, frame.chunkSize);
-        int lastFrameIndex = fs.size() - 1;
-        int endChunkIndex = startFrameIndex + frame.chunkSize - 1;
-        if (endChunkIndex > lastFrameIndex) {
-            endChunkIndex = lastFrameIndex;
-        }
 
+        FrameSet chunkFrameSet = new FrameSet(frameSpec);
+        int chunkEndFrame = chunkFrameSet.get(-1);
```

Comment:
Hey @anton-ubi, thanks for this fix :)
However I got an error there when building the docker image:
`java.lang.IndexOutOfBoundsException: index (-1) must not be negative`
I guess we can't access negative indices in java.
```suggestion
        int chunkEndFrame = chunkFrameSet.get(chunkFrameSet.size()-1);
```